### PR TITLE
Only modify default Application and JUnit run configs in BaselineIdea

### PR DESCRIPTION
**Pull request by henryptung**
Don't mutate existing non-default configs.

### CODE REVIEW

**Comment by henryptung** [comment]
Targets #162

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy [hunk 1]
```diff
@@ -164,13 +164,14 @@ class BaselineIdea extends AbstractBaselinePlugin {
...
 
         if (runManager) {
-            // if RunManager configuration exists, set all WORKING_DIRECTORY options to module
-            def items = runManager.configuration.option.findAll { it.'@name' == 'WORKING_DIRECTORY' }
-            items.each { item ->
-                item.'@value' = 'file://$MODULE_DIR$'
-            }
+            // if RunManager configuration exists, set Application, JUnit WORKING_DIRECTORY default to $MODULE_DIR$
+            def appJunitDefaults = new NodeList(runManager.configuration
+                    .findAll { it.'@default' == 'true' && it.'@type' in ['Application', 'JUnit'] })
```
Comment (uschi2000) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy]: can you pull out a variable `def configurations = ['Application', 'JUnit']` and use it here and in the else block?

**Review by uschi2000 — COMMENTED** [review]
one nitpick, otherwise +1. thanks, Henry!

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy (hunk 1 above)
Comment (henryptung) [reply gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy]: Hmm, if we're talking about code duplication, how do you feel about this?

```
    /**
     * Configure the default working directory of RunManager configurations to be the module directory.
     */
    private static void setRunManagerWorkingDirectory(Node node) {
        def runTypes = ['Application', 'JUnit'] as Set

        def runManager = matchOrCreateChild(node, 'component', [name: 'RunManager'])
        runTypes.each { runType ->
            def configuration = matchOrCreateChild(runManager, 'configuration',
                    [default: 'true', type: runType],
                    [factoryName: runType])
            def workingDirectory = matchOrCreateChild(configuration, 'option', [name: 'WORKING_DIRECTORY'])
            workingDirectory.'@value' = 'file://$MODULE_DIR$'
        }
    }

    private static Node matchOrCreateChild(Node base, String name, Map attributes = [:], Map defaults = [:]) {
        def child = base[name].find { it.attributes().entrySet().containsAll(attributes.entrySet()) }
        if (child) {
            return child
        }

        return base.appendNode(name, attributes + defaults)
    }
```

Comment (uschi2000) [reply gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy]: yep, that's cool

Comment (henryptung) [reply gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineIdea.groovy]: pushed fix, thanks!

**Review by uschi2000 — APPROVED** [review]
cheers, Henry.

**Pull request merged**